compiler/src/iree/compiler/GlobalOptimization/Interfaces/HoistableTypeInterface.cpp

```diff
@@ -85,6 +86,34 @@ struct HoistableTensorTypeInterface
   }
 };
 
+struct HoistableIndexTypeInterface
+    : public IREE::Util::HoistableTypeInterface::ExternalModel<
+          HoistableIndexTypeInterface, IndexType> {
+  bool isHoistableType(Type type) const { return type.isIndex(); }
+  bool isHoistableLeafType(Type type) const { return type.isIndex(); }
```

Comment:
Since this is an interface for the IndexType, `type.isIndex()` will always be true, so you can just make these `return true`.